lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -137,20 +131,59 @@ export abstract class ClientApplication {
         };
     }
 
+    /**
+     * Generates a request with the default scopes.
+     * @param request 
+     */
+    protected generateUrlRequest(request: AuthorizationUrlRequest): AuthorizationUrlRequest {
```

Comment:
We should not duplicate this code here (ln 134-175) and device code request in PublicClientApplication. 

Consider creating ScopeSet and passing in as part of ClientConfiguration or creating a parent class for requests.